app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    def wrapper(*args: Any) -> Any:
+        if args in cache_store:
+            print("Getting from cache")
+            return cache_store[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_store[args] = result
+            return result
```

Comment:
you don't need else because if statement has return, it means it will not reach the code after it :)